src/MakingSense.AspNetCore.Documentation/DocumentationMiddleware.cs

```diff
@@ -179,5 +220,36 @@ private bool TryGetDocumentsSubpathAndLanguage(HttpRequest request, out PathStri
 			return true;
 		}
 
+		private void ConfigureNotFoundFiles()
+		{
+			_notFoundHtmlFile = _options.NotFoundHtmlPath == null && _options.NotFoundHtmlFile != null ?
+				_options.NotFoundHtmlFile : _options.FileProvider.GetFileInfo(_options.NotFoundHtmlPath);
+
+			if (_options.NotFoundHtmlPath != null && _notFoundHtmlFile.Exists)
+			{
+				_notFoundHtmlFilesPerLanguage = GetFileInfoByLanguage(_options.NotFoundHtmlPath);
+			}
+		}
+
+		private Dictionary<string, IFileInfo> GetFileInfoByLanguage(string path)
+		{
+			var fileInfoByLang = new Dictionary<string, IFileInfo>();
+
+			foreach (var lang in _options.SupportedLanguages)
+			{
+				var pathWithoutExt = path.Substring(0, path.LastIndexOf('.'));
+				var ext = path.Substring(path.LastIndexOf('.'));
+
+				var fileInfo = _options.FileProvider.GetFileInfo(pathWithoutExt + $".{lang}" + ext);
```

Comment:
I think that it would be clever to use only a kind of string composition way, for example: `pathWithoutExt + "." + lang + ext` or `$"{pathWithoutExt}.{lang}{ext}"`.